x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichProcessorFactory.java

```diff
@@ -36,12 +36,14 @@
     static final String TYPE = "enrich";
     private final Client client;
     private final ScriptService scriptService;
+    private final EnrichCache enrichCache;
 
     volatile Metadata metadata;
 
-    EnrichProcessorFactory(Client client, ScriptService scriptService) {
+    EnrichProcessorFactory(Client client, ScriptService scriptService, EnrichCache enrichCache) {
         this.client = client;
         this.scriptService = scriptService;
+        this.enrichCache = enrichCache;
```

Comment:
Maybe require non-null here since the cache is used unconditionally in `createSearchRunner`?

Reply:
👍 will do that

Reply:
Fixed: 9deb1b54da6416b062971da71c3f09bdb379accf